File: max-consecutive-ones/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution: O(n) time, O(1) additional space.
        count = 0
        max_count = 0
        
        for num in nums:
            if num == 1:
                count += 1
                max_count = max(max_count, count)
            else:
                count = 0
                
        return max_count
                
                

File: max-consecutive-ones/max-consecutive-ones.java
class Solution {
    
    public int findMaxConsecutiveOnes(int[] nums) {
        
        // Solution: O(n) time, O(1) additional space.
        
        int count = 0;
        int maxCount = 0;
        
        for (int num: nums) {
            
            if (num == 1) {
                count += 1;
                maxCount = Math.max(maxCount, count);
            } else {
                count = 0;
            }
            
        }
        
        return maxCount;

    }
    
}